homeassistant/components/dreo/fan.py

```diff
@@ -0,0 +1,173 @@
+        return self._attr_state
+
+    @cached_property
+    def preset_modes(self) -> list[str]:
+        """Return a list of available preset modes.
+
+        Requires FanEntityFeature.SET_SPEED.
+        """
+        if hasattr(self, "_attr_preset_modes"):
+            return self._attr_preset_modes
+        return FAN_DEVICE.get("config").get(self._model).get("preset_modes")
+
+    @cached_property
+    def preset_mode(self) -> str | None:
+        """Return the current preset mode, e.g., auto, smart, interval, favorite.
+
+        Requires FanEntityFeature.SET_SPEED.
+        """
+        if hasattr(self, "_attr_preset_mode"):
+            return self._attr_preset_mode
+        return None
+
+    @cached_property
+    def speed_count(self) -> int:
+        """Return the number of speeds the fan supports."""
+        if hasattr(self, "_attr_speed_count"):
+            return self._attr_speed_count
+        return int_states_in_range(
+            FAN_DEVICE.get("config").get(self._model).get("speed_range")
+        )
+
+    @cached_property
+    def percentage(self) -> int | None:
+        """Return the current speed as a percentage."""
+        if hasattr(self, "_attr_percentage"):
+            return ranged_value_to_percentage(
+                FAN_DEVICE.get("config").get(self._model).get("speed_range"),
+                self._attr_percentage
+            )
+        return 0
+
+    @cached_property
+    def oscillating(self) -> bool | None:
+        """Return whether the fan is currently oscillating."""
+        return self._attr_oscillating
+
+    @property
+    def available(self) -> bool:
+        """Return True if device is available."""
+        return self._attr_available
+
+    def turn_on(self, **kwargs: Any) -> None:
+        """Turn the device on."""
+        result = self._try_command(
+            "Turn the device on failed.", power_switch=True
+        )
+
+        if result:
+            self._state = True
+
+    def turn_off(self, **kwargs: Any) -> None:
+        """Turn the device off."""
+        result = self._try_command(
+            "Turn the device off failed.", power_switch=False
+        )
+
+        if result:
+            self._state = False
+
+    def set_preset_mode(self, preset_mode: str) -> None:
+        """Set the preset mode of fan."""
+        result = self._try_command(
+            "Set the preset mode failed.", mode=preset_mode
+        )
+
+        if result:
+            self._mode = preset_mode
+
+    def set_percentage(self, percentage: int) -> None:
+        """Set the speed of fan."""
+        speed = math.ceil(
+            percentage_to_ranged_value(
+                FAN_DEVICE.get("config").get(self._model).get("speed_range"), percentage
+            )
+        )
+
+        if speed == 0:
+            return
+
+        result = self._try_command(
+            "Set the speed failed.", speed=speed
+        )
+
+        if result:
+            self._speed = speed
```

Comment:
result is always `true`

```suggestion
        self._try_command(
            "Set the speed failed.", speed=speed
        )
        self._speed = speed
```